homeassistant/components/mystrom/__init__.py

```diff
@@ -1 +1,78 @@
-"""The mystrom component."""
+"""The myStrom integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+from pymystrom.bulb import MyStromBulb
+from pymystrom.exceptions import MyStromConnectionError
+from pymystrom.switch import MyStromSwitch
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def get_device_info(host: str) -> dict[str, Any]:
+    """Get the device info of a myStrom device."""
+    async with aiohttp.ClientSession() as session:
+        async with session.get(f"http://{ host}/api/v1/info") as response:
```

Comment:
Newer devices are using a different endpoint. See https://github.com/home-assistant-ecosystem/python-mystrom/pull/34

Reply:
I'm not sure whether I get your point: In the referenced PR the endpoint is `api/v1/info`. In the [mystrom documentation](https://api.mystrom.ch/#f37a4be7-0233-4d93-915e-c6f92656f129) the endpoint is `api/v1/info`.

What is the specific change you propose?

Reply:
Depending on the firmware there are two different endpoints available to get the details about the switch. 

- Older firmware: `info.json`
- New firmware: `api/v1/info`

Reply:
Implemented both versions